tests/integration/framework/process/daprd/options.go

```diff
@@ -305,6 +305,12 @@ func WithAppAPIToken(t *testing.T, token string) Option {
 	))
 }
 
+func WithAppAPITokenHeader(t *testing.T, token string) Option {
+	return WithExecOptions(exec.WithEnvVars(t,
```

Comment:
Hey @slimjim777 thanks for the PR 🎉 mind adding an integration test using this function to test the logic? You can likely cherry-pick the test from this [old PR here](https://github.com/dapr/dapr/pull/7190/files#diff-8b35b56824d91b4056d86fc262afb5bf0c31ea12ac1809a9434fb238b65ad73a) - please keep in mind the comments from that test :) 

Additionally, can you please open a PR in the dapr/docs repo to document this?